backend/src/main/java/ai/verta/modeldb/advancedService/AdvancedServiceImpl.java

```diff
@@ -554,30 +565,19 @@ public void getHydratedExperimentRunsInProject(
         LOGGER.trace("comments {}", comments);
         hydratedExperimentRunBuilder.addAllComments(comments);
 
-        List<Action> actionList =
-            ModelDBUtils.getActionsList(new ArrayList<>(projectIdSet), actions);
-        LOGGER.info("actionList {}", actionList);
-        Action deleteAction =
-            Action.newBuilder()
-                .setModeldbServiceAction(ModelDBServiceActions.DELETE)
-                .setService(Service.MODELDB_SERVICE)
-                .build();
-        Action updateAction =
-            Action.newBuilder()
-                .setModeldbServiceAction(ModelDBServiceActions.UPDATE)
-                .setService(Service.MODELDB_SERVICE)
-                .build();
         LOGGER.info(
             "roleService.isCurrentUser(experimentRun.getOwner() {}",
             authService.isCurrentUser(experimentRun.getOwner()));
-        if (currentUserVertaID.equalsIgnoreCase(experimentRun.getOwner())
-            && !actionList.contains(deleteAction)
-            && actionList.contains(updateAction)) {
-          actionList.add(deleteAction);
+        if (currentUserVertaID.equalsIgnoreCase(experimentRun.getOwner())) {
+          List<Action> experimentRunActionList =
+              experimentRunActions.get(experimentRun.getId()).getActionsList();
+          LOGGER.debug("ExperimentRun Owner actionList {}", experimentRunActionList);
+          hydratedExperimentRunBuilder.addAllAllowedActions(experimentRunActionList);
```

Comment:
we should add  projectActionList outside the if .

Reply:
but `hydratedExperimentRunBuilder.addAllAllowedActions` contains only action type and we do not identify that action is for project or Experiment? so I have added `IF`. if ExperimentRun action found then set those actions and if not found then add Project actions in that list.